lib/gpio.cc

```diff
@@ -269,10 +269,26 @@ static uint32_t ReadRevisionFromProcCpuinfo() {
   return pi_revision;
 }
 
+static uint32_t read_be32(const uint8_t *p) {
+  return p[0] << 24 | p[1] << 16 | p[2] << 8 | p[3];
+}
+
+static uint32_t ReadRevisionFromDeviceTree() {
+  char buffer[4];
+  if (ReadFileToBuffer(buffer, sizeof(buffer), "/proc/device-tree/system/linux,revision") < 0) {
+    fprintf(stderr, "Failed to read revision from /proc/device-tree\n");
+    return read_be32((const uint8_t *)buffer);
```

Comment:
Another thing here: I've used a C-style cast, but I'm open to suggestions if you think there's a C++ cast that fits better